corehq/apps/users/management/commands/remove_mobile_permissions.py

```diff
@@ -0,0 +1,36 @@
+from django.core.management.base import BaseCommand
+
+from corehq.apps.domain.models import Domain
+from corehq.apps.users.models_role import RolePermission, UserRole
+from corehq.toggles import RESTRICT_MOBILE_ACCESS
+
+
+class Command(BaseCommand):
+    help = "Removes the mobile access permission from roles not in Domains with" \
+           " the RESTRICT_MOBILE_ACCESS feature flag"
+
+    def handle(self, **options):
+        num_domains_with_their_roles_changed = 0
+        domains_with_restriction_on = RESTRICT_MOBILE_ACCESS.get_enabled_domains()
+        for domain in Domain.get_all_names():
+            if domain not in domains_with_restriction_on and not Domain.get_by_name(domain).restrict_mobile_access:
+                permission_was_removed_for_domain = None
+                for role in UserRole.objects.get_by_domain(domain):
```

Comment:
The challenge with that approach is that if the user running the command wants to save the log they would need to pipe the output to a file. If they forget then the output could be lost. The purpose of this logging is to create persistent record of the actions in case they need to be referred to later and I would recommend that once the command is run the log be attached to the Jira ticket for reference.

I would recommend writing the logs as a CSV file since that makes them easier to process if needed. Here's an example: https://github.com/dimagi/commcare-hq/blob/5a0a6be2ff5851c830196f70504b826c88fd2011/corehq/apps/auditcare/management/commands/export_domain_logins.py#L61-L70